pkg/server/conn.go

```diff
@@ -2581,3 +2599,41 @@ func (cc *clientConn) ReadPacket() ([]byte, error) {
 func (cc *clientConn) Flush(ctx context.Context) error {
 	return cc.flush(ctx)
 }
+
+// Stats returns the connection statistics.
+func (*clientConn) Stats(vars *variable.SessionVars) (map[string]interface{}, error) {
+	m := make(map[string]interface{}, len(clientConnStatus))
+	for name, v := range clientConnStatus {
+		m[name] = v.Value
+	}
+
+	switch vars.CompressionAlgorithm {
+	case mysql.CompressionNone:
+		m[statusCompression] = "OFF"
+		m[statusCompressionAlgorithm] = ""
+		m[statusCompressionLevel] = 0
+	case mysql.CompressionZlib:
+		m[statusCompression] = "ON"
+		m[statusCompressionAlgorithm] = "zlib"
+		m[statusCompressionLevel] = mysql.ZlibCompressDefaultLevel
+	case mysql.CompressionZstd:
+		m[statusCompression] = "ON"
+		m[statusCompressionAlgorithm] = "zstd"
+		m[statusCompressionLevel] = vars.CompressionLevel
+	default:
+		logutil.Logger(context.Background()).Debug(
```

Comment:
seems it's equal to log.L()

Reply:
Like this?

Reply:
Or something like this?
```patch
diff --git a/pkg/server/conn.go b/pkg/server/conn.go
index 4518be086b..891fba09b3 100644
--- a/pkg/server/conn.go
+++ b/pkg/server/conn.go
@@ -2621,7 +2621,7 @@ func (*clientConn) Stats(vars *variable.SessionVars) (map[string]interface{}, er
                m[statusCompressionAlgorithm] = "zstd"
                m[statusCompressionLevel] = vars.CompressionLevel
        default:
-               logutil.Logger(context.Background()).Debug(
+               logutil.BgLogger().Debug(
                        "unexpected compression algorithm value",
                        zap.Int("algorithm", vars.CompressionAlgorithm),
                )
```

Reply:
Oh I think the two changes are both OK

Reply:
Pushed in 5aca0b2f7ce6d89c2e5dcb0e4b581a02e75dbbfa